manifests/network/interface.pp

```diff
@@ -100,7 +103,9 @@
       }
 
       bsd_interface { $if_name:
-        ensure => $ensure,
+        ensure  => $ensure,
+        parents => $parents,
+        require => Shell_config[$rec_hash],
```

Comment:
Its probably not just the shell_config we want to require here.  I'm thinking something like:
```Puppet
bsd_interface { $if_name:
...
  require => Bsd_interface[$parents],
}
```
This way we ensure that the entire thing is required, rather than just a component inside the interface, to make sure the interface is up and addressed etc.